src/main/java/frc/robot/hardware/phoenix6/signal/Phoenix6SignalBuilder.java

```diff
@@ -0,0 +1,62 @@
+package frc.robot.hardware.phoenix6.signal;
+
+import com.ctre.phoenix6.StatusSignal;
+import frc.utils.AngleUnit;
+import frc.robot.hardware.phoenix6.Phoenix6Utils;
+
+public class Phoenix6SignalBuilder {
+
+	private static final int UPDATE_FREQUENCY_RETRIES = 5;
+
+	private static void setFrequencyWithRetry(StatusSignal<?> signal, double frequency) {
+		Phoenix6Utils.checkWithRetry(() -> signal.setUpdateFrequency(frequency), UPDATE_FREQUENCY_RETRIES);
+	}
+
+	private static StatusSignal<?> cloneWithFrequency(StatusSignal<?> signal, double frequency) {
+		StatusSignal<?> signalClone = signal.clone();
+		setFrequencyWithRetry(signalClone, frequency);
+		return signalClone;
+	}
+
+	public static Phoenix6DoubleSignal generatePhoenix6Signal(StatusSignal<?> signal, double frequency) {
+		StatusSignal<?> signalClone = cloneWithFrequency(signal, frequency);
+		return new Phoenix6DoubleSignal(signalClone.getName(), signalClone);
+	}
+
+	public static Phoenix6AngleSignal generatePhoenix6Signal(StatusSignal<?> signal, double frequency, AngleUnit angleUnit) {
+		StatusSignal<?> signalClone = cloneWithFrequency(signal, frequency);
+		return new Phoenix6AngleSignal(signalClone.getName(), signalClone, angleUnit);
+	}
+
+	public static Phoenix6LatencySignal generatePhoenix6Signal(
+		StatusSignal<?> signal,
+		SignalGetter signalSlope,
+		double frequency,
+		AngleUnit angleUnit
+	) {
+		StatusSignal<?> signalClone = cloneWithFrequency(signal, frequency);
+		setFrequencyWithRetry(signalSlope.getSignal(), frequency);
+		return new Phoenix6LatencySignal(signalClone.getName(), signalClone, signalSlope.getSignal(), angleUnit);
+	}
+
+	/**
+	 * Use only if not fetching the slope signal!!!
+	 */
+	public static Phoenix6BothLatencySignal generatePhoenix6Signal(
```

Comment:
This is a bit risky

Reply:
I don’t have a solution for that. 
Using the this class if fetching the other would not make something bad, just less better by a bit.
Also the way of creating signals (the builder) 
Make it easier to create the right signal or of the 4

Reply:
A "fix" can also be by simply changing the name
When I will use this function, I will read the function name, but docstring, not so sure.